apps/tufctl/main.cpp

```diff
@@ -0,0 +1,94 @@
+#include <boost/property_tree/ini_parser.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/ptree_fwd.hpp>
+#include "aktualizr-lite/tuf/tuf.h"
+
+int main(int argc, char** argv) {
+  if (argc < 2) {
+    std::cerr << "Usage example: " << argv[0] << " repo_sources.toml" << std::endl;
+    exit(1);
+  }
+
+  std::vector<std::shared_ptr<aklite::tuf::TufRepoSource>> sources;
+  // Set up
+  {
+    boost::property_tree::ptree pt;
+    boost::property_tree::ini_parser::read_ini(argv[1], pt);
+
+    std::ostringstream oss;
+    boost::property_tree::write_json(oss, pt);
+    std::cout << oss.str();
+
+    for (boost::property_tree::ptree::iterator pos = pt.begin(); pos != pt.end();) {
+      std::cout << pos->first << std::endl;
+
+      if (pos->first.rfind("source ") == 0) {
+        std::cout << "got repo " << pos->first.substr(7) << std::endl;
+        std::string uri = pos->second.get<std::string>("uri");
+        std::cout << "uri " << uri << " " << uri.rfind("\"file://", 0) << std::endl;
+
+        std::shared_ptr<aklite::tuf::TufRepoSource> source;
+        if (uri.rfind("\"file://", 0) == 0) {
+          source = std::make_shared<aklite::tuf::AkLocalTufRepoSource>(pos->first, pos->second);
+        } else {
+          source = std::make_shared<aklite::tuf::AkHttpsTufRepoSource>(pos->first, pos->second);
+        }
+        sources.push_back(source);
+      }
+
+      ++pos;
+    }
+  }
+
+  // Try individual fetch operations. sota.toml is not used
+  {
+    for (auto const& source : sources) {
+      Json::StreamWriterBuilder builder;
+      builder["indentation"] = "  ";
+      try {
+        auto json = source->fetchRoot(1);
```

Comment:
I'd expect a loop to fetch all root roles here, starting at the version _currently stored locally plus 1_, and ending at the _version for which we received 404_.

Reply:
This is just a test main. That logic should be in a real client that uses the API. Also, it should support importing of the initial TUF root meta - which is essentially update from a local source TUF repo.